test/unit/orchestration_test.rb

```diff
@@ -51,6 +51,49 @@ def setTest; true; end
     end
   end
 
+  test "test host compensates queue if active record not saved" do
+    class Host::Test < Host::Base
```

Comment:
Perhaps avoid reusing the same class name as is defined above, the class definitions will be composed together.